src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -76,6 +76,13 @@ private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appCo
 
             if (featureDefinition != null)
             {
+                //
+                // Default to true for requirement type All
+                if (featureDefinition.RequirementType == RequirementType.All)
+                {
+                    enabled = true;
+                }
+
                 //
```

Comment:
The `Any` check in

`                if (featureDefinition.EnabledFor.Any(featureFilter => string.Equals(featureFilter.Name, "AlwaysOn", StringComparison.OrdinalIgnoreCase)))
`

Doesn't work for requirementtype.all

Reply:
Do you mean it shouldn't work, or in it's current state, it doesn't work?

Reply:
I guess both. Right now the presence of the AlwaysOn filter will short circuit the evaluation and enable the feature. Even for `all`